wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/RecipeFileContentProvider.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.devfile.server;
+
+/**
+ * Fetches content of file described in local field of recipe-type {@link
+ * org.eclipse.che.api.devfile.model.Tool}
+ *
+ * @author Max Shaposhnyk
+ */
+@FunctionalInterface
+public interface RecipeFileContentProvider {
+
+  String fetchContent(String localFileName);
```

Comment:
Never heard that is should be a path. It's a devfile-relative filename IMO.

Reply:
>  It's a devfile-relative filename IMO.

Do you mean devfile-relative file path?

Reply:
I may be wrong but for me filename is like `petclinic.yaml`, and `../app/petclinic.yaml` is file path

Reply:
No - filename. It can be `petclinic.yaml` but not sure if we should support also to be `/foo/bar/petclinic.yaml` (btw it currelnty may work OK too :) )

Reply:
`../app/petclinic.yaml` definitely will not work

Reply:
As far as I see it is root a repository-relative path. BTW Please add a clear description for this argument.

Reply:
Ok expanded description